src/en/kmanga/src/eu/kanade/tachiyomi/extension/en/kmanga/KManga.kt

```
@@ -131,30 +119,69 @@ class KManga : HttpSource() {
+            val url = apiUrl.toHttpUrl().newBuilder()
+                .addPathSegments("search/title")
+                .addQueryParameter("keyword", query)
+                .addQueryParameter("offset", offset.toString())
+                .addQueryParameter("limit", searchLimit.toString())
+                .build()
+            return hashedGet(url)
+        }
 
-        return when {
-            query.isNotBlank() -> GET("$baseUrl/search/$query", headers)
-            genreFilter.state != 0 -> GET("$baseUrl${genreFilter.toUriPart()}", headers)
-            else -> popularMangaRequest(page)
+        if (genreFilter.state != 0) {
+            val url = baseUrl.toHttpUrl().newBuilder()
+                .addPathSegments(genreFilter.toUriPart().removePrefix("/"))
+                .build()
+            return GET(url, headers)
         }
+        return popularMangaRequest(page)
     }
 
     override fun searchMangaParse(response: Response): MangasPage {
-        val requestUrl = response.request.url.toString()
-        return if (requestUrl.contains("/search/") || requestUrl.contains("/search/genre/")) {
-            val document = response.asJsoup()
-            val mangas = document.select("ul.c-search-items li.c-search-items__item a.c-search-item").map(::searchMangaFromElement)
-            MangasPage(mangas, false)
-        } else {
-            popularMangaParse(response)
+        if (response.request.url.host.contains("api.")) {
+            return try {
+                val result = response.parseAs<SearchApiResponse>()
+                val mangas = result.titleList.map { manga ->
+                    SManga.create().apply {
+                        url = "/title/${manga.titleId}"
+                        title = manga.titleName
+                        thumbnail_url = manga.thumbnailImageUrl
+                    }
+                }
+                MangasPage(mangas, mangas.size >= searchLimit)
+            } catch (e: Exception) {
```

no need to catch exceptions here, better to surface the error